src/scripts/main.js

```diff
@@ -1,3 +1,242 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('table');
+let sortType = 'ASC';
+let lastSortedTitle = null;
+let editingCell = null;
+
+document.querySelector('thead').addEventListener('click', (e) => {
+  const headerTitles = [...table.tHead.rows[0].cells];
+  const clickedTitle = headerTitles.find((el) => el === e.target);
+
+  if (!clickedTitle) {
+    return;
+  }
+
+  const index = headerTitles.indexOf(clickedTitle);
+  const tbody = table.querySelector('tbody');
+  const rows = [...tbody.rows];
+
+  if (lastSortedTitle === index) {
+    sortType = sortType === 'ASC' ? 'DESC' : 'ASC';
+  } else {
+    sortType = 'ASC';
+    lastSortedTitle = index;
+  }
+
+  rows.sort((a, b) => {
+    const cellA = a.cells[index].textContent.trim();
+    const cellB = b.cells[index].textContent.trim();
+
+    if (/\$/.test(cellA)) {
+      const numA = parseFloat(cellA.replace(/[^\d.-]/g, ''));
+      const numB = parseFloat(cellB.replace(/[^\d.-]/g, ''));
+
+      return sortType === 'ASC' ? numA - numB : numB - numA;
+    }
+
+    return sortType === 'ASC'
+      ? cellA.localeCompare(cellB)
+      : cellB.localeCompare(cellA);
+  });
+
+  tbody.innerHTML = '';
+  rows.forEach((row) => tbody.appendChild(row));
+});
+
+document.querySelector('tbody').addEventListener('click', (e) => {
+  [...table.querySelector('tbody').rows].forEach((row) => {
+    if (row.contains(e.target)) {
+      row.classList.add('active');
+    } else {
+      row.classList.remove('active');
+    }
+  });
+});
+
+(function addedEmployeeForm() {
+  const form = document.createElement('form');
+  const selectValues = [
+    'Tokyo',
+    'Singapore',
+    'London',
+    'New York',
+    'Edinburgh',
+    'San Francisco',
+  ];
+
+  form.classList.add('new-employee-form');
+
+  const dataParam = [...table.tHead.rows[0].cells].map((el) => {
+    return el.textContent.toLowerCase();
+  });
+
+  dataParam.forEach((el) => addNeededInput(el));
+
+  const submitButton = document.createElement('button');
+
+  submitButton.textContent = 'Save to table';
+  submitButton.type = 'submit';
+  form.appendChild(submitButton);
+
+  document.body.appendChild(form);
+
+  form.addEventListener('submit', (e) => {
+    e.preventDefault();
+
+    const newEmployeeData = {};
+    const formData = new FormData(form);
+
+    formData.forEach((value, key) => {
+      if (key === 'salary') {
+        newEmployeeData[key] = `$${parseFloat(value).toLocaleString('en-US')}`;
+      } else {
+        newEmployeeData[key] = value;
+      }
+    });
+
+    const message = validateEmployeeData();
+
+    notification(message);
```

Comment:
The `notification` function is called with `message`, which might be `null` if validation passes. Ensure that `message` is defined before calling `notification` to avoid potential issues.